File: 1-easy/problems/3074-apple-redistribution-into-boxes/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def minimumBoxes(self, apple, capacity):
        """
        :type apple: List[int]
        :type capacity: List[int]
        :rtype: int
        """
        capacity.sort(reverse=True)
        total = sum(apple)
        for i in range(len(capacity)):
            total -= capacity[i]
            if total <= 0:
                return i+1
        return -1

File: 1-easy/problems/3074-apple-redistribution-into-boxes/solution.ts
function minimumBoxes(apple: any, capacity: any): boolean | number | string | any {
  """
  :type apple: List[int]
  :type capacity: List[int]
  :rtype: int
  """
  capacity.sort(reverse=true)
  total = sum(apple)
  for (i in Array.from({length: (capacity}, (_, i) => i).length)) {
      total -= capacity[i]
      if (total <= 0) {
          return i+1;
  return -1;
}

export { minimumBoxes };
